src/services/nunjucks.js

```diff
@@ -28,51 +30,79 @@ function createEnvironment (context) {
     return '<pre><code>' + string + '</code></pre>';
   });
 
-  addPlugins(env, context);
-
-  return env;
+  addPlugins(env, context, function (err, env) {
+    callback(err, env);
+  });
 }
 
-var addPlugins = function (env, context) {
+var addPlugins = function (env, context, callback) {
   var pluginPath = services.path.getPluginsPath(context);
   try {
     fs.openSync(pluginPath, 'r');
   } catch(e) {
     logger.warn('Unable to find plugins directory at: ' + pluginPath);
-    return;
+    return callback(null, env);
   }
 
   if (!fs.statSync(pluginPath).isDirectory()) {
-    return;
+    return callback(null, env);
   }
 
-  fs.readdirSync(pluginPath).forEach(function (pluginDir) {
-    var plugin;
+  async.eachSeries(fs.readdirSync(pluginPath), function (pluginDir, callback) {
```

Comment:
I bet this could be done in parallel (`async.each`); because each directory gets its own `node_modules`, the npm operations in each shouldn't interfere with one another. :rocket: